### PR TITLE
Integration tests for vertx-opentracing with Jaeger testcontainers

### PR DESCRIPTION
**Pull request by jcarranzan**
Adding integration tests for vertx-opentracing using Jaeger testcontainers.

### CODE REVIEW

File: vertx-opentracing/pom.xml [hunk 1]
```diff
@@ -56,6 +48,46 @@
...
+      <artifactId>rest-assured</artifactId>
+      <version>5.2.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.jaegertracing</groupId>
+      <artifactId>jaeger-core</artifactId>
+      <version>1.8.1</version>
+    </dependency>
+    <dependency>
```
Comment (vietj) [inline vertx-opentracing/pom.xml:67]: why is this dependency on itself needed ?

Comment (jcarranzan) [reply vertx-opentracing/pom.xml]: This dependency is not needed, I've removed it. Thanks @vietj .

File: vertx-opentracing/pom.xml [hunk 2]
```diff
@@ -32,18 +32,10 @@
   </properties>
 
   <dependencies>
-
-    <dependency>
-      <groupId>io.jaegertracing</groupId>
-      <artifactId>jaeger-client</artifactId>
-      <version>1.0.0</version>
-    </dependency>
```
Comment (tsegismont) [inline vertx-opentracing/pom.xml:40]: Why has this been removed?

File: vertx-opentracing/src/test/java/io/vertx/tracing/opentracing/it/BasicTracingIT.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      tracer = JAEGER_ALL_IN_ONE.createTracer(JAEGER_SERVICE_NAME);
+      deployCheck.flag();
+
+      tracedVertx = Vertx.vertx(
+        new VertxOptions().setTracingOptions(
+          new OpenTracingOptions(tracer))
+      );
+
+      tracedVertx.deployVerticle(ServerVerticle.class.getName(), context.succeeding(id -> {
+        await("Await until route is ready!").until(() -> {
```
Comment (tsegismont) [inline vertx-opentracing/src/test/java/io/vertx/tracing/opentracing/it/BasicTracingIT.java:54]: It's not a good practice to use blocking libraries such as `awaitility` on a vert.x event loop. You should either use Vert.x jUnit 5 tools or block in the main test thread (with the help of a `CountDownLatch`).

**Review by tsegismont — COMMENTED** [review]
Thank you @jcarranzan

File: vertx-opentracing/src/test/java/io/vertx/tracing/opentracing/it/BasicTracingIT.java (hunk 3 above)
Comment (jcarranzan) [reply vertx-opentracing/src/test/java/io/vertx/tracing/opentracing/it/BasicTracingIT.java]: Ok I see it, I will use vert.x junit5 tools, thank you @tsegismont !

**Comment by jcarranzan** [comment]
@tsegismont do you think is it ready to merge? thanks!

**Comment by tsegismont** [comment]
@jcarranzan have you changed anything since the last review?

**Comment by jcarranzan** [comment]
Oh sorry I did squash my commits and force push then I think you can't see only my last changes, you see all the changes again.  Well, I did the improvements according to your suggestions, for instance I deleted  awaitility library use by using as we do in vertx junit 5 --> https://github.com/eclipse-vertx/vertx-tracing/blob/e8ba13e853ecc6afbf9b47580b820ae3763089e0/vertx-opentracing/src/test/java/io/vertx/tracing/opentracing/it/BasicTracingIT.java#L56 
The only thing that I haven't changed about your comments is the unresolved thread comment.

File: pom.xml [hunk 4]
```diff
@@ -44,6 +44,13 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>org.testcontainers</groupId>
+        <artifactId>testcontainers-bom</artifactId>
+        <version>1.17.5</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
```
[the remark this answers is not in the document]
Comment (tsegismont) [reply pom.xml:55]: The problem is we have to check the testcontainers bom manually to verify if it doesn't force dependency versions for anything else that what we actually use. And it applies to runtime scope...

I believe it's safer if we add a couple Maven properties and set versions manually.

**Review by tsegismont — COMMENTED** [review]
Thank you for the updates, it all looks good to me, except for the problem with the testcontainers BOM.

**Comment by tsegismont** [comment]
@jcarranzan it seems like the tests is not able to start because nothing is listening on port 8080

File: pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jcarranzan) [reply pom.xml]: Ok @tsegismont , I did the related changes here -> https://github.com/eclipse-vertx/vertx-tracing/pull/48/commits/c0095a30a3af55cceea37cbd9c57db046168858e

**Comment by jcarranzan** [comment]
I am not sure why that connection refuse occurred because both tests use ServerVerticle and it was listening on port 8080, I've added promise logic on the server now. Can you run again the CI to see if it keeps happening?
https://github.com/eclipse-vertx/vertx-tracing/pull/48/commits/62611299eea7befc5d3f08398ee23b8dc7f10cdd

File: pom.xml [hunk 5]
```diff
@@ -33,6 +35,8 @@
     <stack.version>4.3.5-SNAPSHOT</stack.version>
     <doc.skip>true</doc.skip>
     <jar.manifest>${project.basedir}/src/main/resources/META-INF/MANIFEST.MF</jar.manifest>
+    <testcontainers.version>1.17.5</testcontainers.version>
+    <pgcontainer.version>1.15.2</pgcontainer.version>
```
Comment (tsegismont) [inline pom.xml:39]:
```suggestion
    <testcontainers.version>1.15.2</testcontainers.version>
```

File: pom.xml [hunk 6]
```diff
@@ -76,9 +80,14 @@
     <dependency>
       <groupId>org.testcontainers</groupId>
       <artifactId>postgresql</artifactId>
-      <version>1.15.2</version>
+      <version>${pgcontainer.version}</version>
```
Comment (tsegismont) [inline pom.xml:83]:
```suggestion
      <version>${testcontainers.version}</version>
```

File: pom.xml [hunk 7]
```diff
@@ -76,9 +80,14 @@
...
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>testcontainers</artifactId>
+      <version>${testcontainers.version}</version>
```
Comment (tsegismont) [inline pom.xml:89]:
```suggestion
      <version>${testcontainers.version}</version>
```

**Review by tsegismont — COMMENTED** [review]
Have you tried to use the same version for testcontainers? It seems `JaegerContainerAllIn` does not require anything specific from the latest version

File: pom.xml (hunk 6 above)
Comment (jcarranzan) [reply pom.xml:83]: Yes, I tried it and works well, but I wasn't sure if you are using that posgresql container version for some specific reason if not, then yes let's use the same as testcontainer version.

**Comment by tsegismont** [comment]
Thank you @jcarranzan !

**Comment by tsegismont** [comment]
Thank you @jcarranzan !

**Pull request merged**